src/engine/test/source/hlp/hlp_test.cpp

```diff
@@ -349,13 +352,64 @@ TEST(hlpTests_map, incomplete_map_test)
     bool ret3 = parseOp(event3, result3);
 
     ASSERT_TRUE(static_cast<bool>(parseOp));
-    ASSERT_TRUE(result1.empty());
+    // ASSERT_TRUE(result1.empty());
     ASSERT_TRUE(result2.empty());
     ASSERT_TRUE(result3.empty());
 }
 
+// MAP: Values before literal
+TEST(hlpTests_map, success_map_1_before_string)
+{
+    const char* logQl = "<_map/kv_map/=/ > hi!";
+    const char* event = "key1=Value1 hi!";
+
+    ParserFn parseOp = getParserOp(logQl);
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
+
+    ParseResult result;
+    bool ret = parseOp(event, result);
+
+    ASSERT_TRUE(static_cast<bool>(parseOp));
+    ASSERT_STREQ(R"({"key1":"Value1"})",
+                 std::any_cast<JsonString>(result["_map"]).jsonString.c_str());
+}
+
+TEST(hlpTests_map, success_map_2_before_string)
+{
+    const char* logQl = "<_map/kv_map/=/ > hi!";
+    const char* event = "key1=Value1 Key2=Value2 hi!";
+
+    ParserFn parseOp = getParserOp(logQl);
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
+
+    ParseResult result;
+    bool ret = parseOp(event, result);
+
+    ASSERT_TRUE(static_cast<bool>(parseOp));
+    ASSERT_STREQ(R"({"key1":"Value1","Key2":"Value2"})",
+                 std::any_cast<JsonString>(result["_map"]).jsonString.c_str());
+}
+
+TEST(hlpTests_map, success_map_multiCharacterArgs)
+{
+    const char* logQl = "<_map/kv_map/: / > hi!";
+    const char* event = "key1: Value1 Key2: Value2 hi!";
+
+    ParserFn parseOp = getParserOp(logQl);
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
+
+    ParseResult result;
+    bool ret = parseOp(event, result);
+
+    ASSERT_TRUE(static_cast<bool>(parseOp));
```

Comment:
why are we testing some things here and others in hlpKVMap_test ?

Reply:
It's true, this shouldn't be here